### DISCUSSION TITLE
github updated post doesnt get marked as unread

### DISCUSSION THREAD

**Discussion by rezad1393** [FreshRSS/FreshRSS#6443] [Q&A]
To Reproduce
add this feed
https://github.com/0x90d/videoduplicatefinder/releases.atom
it has a daily (but it is not daily) build with the same link repeated , like this https://github.com/0x90d/videoduplicatefinder/releases/tag/3.0.x
its post time gets updated when there is a git commit.
so for example last commit was from 17 Apr and the post time got updated to "Tuesday, April 18, 2023 9:08:27 AM"
but in freshrss checking for update doesn't change the post to unread. and it stays on the last time that post was added (or when the feed was added to freshrss) so in my case that is in Feb.

Expected behavior
the post should change to unread.

Environment information (please complete the following information):

    Device: arm odroid xu4

    OS: archlinux-arm

    Browser: firefox

    FreshRSS version: 1.23.1
    php-sqlite 8.3.6-1
    sqlite  sqlite 3.45.3
    PHP version: 8.3.6

    Installation type: self-hosted

when frehsrss has the option to make a post as unread if updated does that include checking for change of time for post too?

**Comment by math-GH** [comment]
![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/203a42d1-b924-4e7a-9bba-12f5bec18a4f)


![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/2176c8da-320d-4428-beb8-8a90c61e2fc0)

**Comment by rezad1393** [comment]
> ![grafik](https://private-user-images.githubusercontent.com/1645099/329206103-203a42d1-b924-4e7a-9bba-12f5bec18a4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNzQ5MDQsIm5iZiI6MTcxNTI3NDYwNCwicGF0aCI6Ii8xNjQ1MDk5LzMyOTIwNjEwMy0yMDNhNDJkMS1iOTI0LTRlN2EtOWJiYS0xMmY1YmVjMThhNGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDlUMTcxMDA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTQ0ZWE4YzFiMWFmMDk5NzBhYjlhYzhhZWU1OTFhZTVlZTdiMTI5M2QyNTM4OTlmODNhOWUxNjBlNjIxMTU2MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dUAa0VZ64BtRDTuWt-1K3wCqADM-mip1C7kzYyempn8)
> 
> ![grafik](https://private-user-images.githubusercontent.com/1645099/329206028-2176c8da-320d-4428-beb8-8a90c61e2fc0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNzQ5MDQsIm5iZiI6MTcxNTI3NDYwNCwicGF0aCI6Ii8xNjQ1MDk5LzMyOTIwNjAyOC0yMTc2YzhkYS0zMjBkLTQ0MjgtYmViOC04YTkwYzYxZTJmYzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDlUMTcxMDA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmYzMmI5MTFkNGVjOTMyMTQ4ODQ0YjVhYWQ2NzM2M2M4MDU0M2YxOTBkYWMyMDM3OTcyM2VjN2EyZTA5NWE2OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ._DxpQAfuqaUWVlFmsLhSd1IfU54ZpQXPtatgNB_8iU4)

tried that.
didnt work.

**Comment by math-GH** [comment]
Oh I see, that you are using FreshRSS 1.21.0. 

Current version is 1.23.1

Maybe it was solve there

**Comment by rezad1393** [comment]
> Oh I see, that you are using FreshRSS 1.21.0.
> 
> Current version is 1.23.1
> 
> Maybe it was solve there

sorry I copied the info from bug report I created last year here https://github.com/FreshRSS/FreshRSS/issues/5347

my installation in current.

**Comment by math-GH** [comment]
Let's do it structured.

This is a snapshot from now:

```
<?xml version="1.0" encoding="UTF-8"?>
<feed xmlns="http://www.w3.org/2005/Atom" xmlns:media="http://search.yahoo.com/mrss/" xml:lang="en-US">
  <id>tag:github.com,2008:https://github.com/0x90d/videoduplicatefinder/releases</id>
  <link type="text/html" rel="alternate" href="https://github.com/0x90d/videoduplicatefinder/releases"/>
  <link type="application/atom+xml" rel="self" href="https://github.com/0x90d/videoduplicatefinder/releases.atom"/>
  <title>Release notes from videoduplicatefinder</title>
  <updated>2021-08-17T11:20:51Z</updated>
  <entry>
    <id>tag:github.com,2008:Repository/164539637/3.0.x</id>
    <updated>2024-05-07T11:38:48Z</updated>
    <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/3.0.x"/>
    <title>3.0.x</title>
    <content type="html">&lt;p&gt;Note: These are daily builds. Attachments are automatically created and replaced on every new commit. Regardless of what the message above says. If you experience upgrade issues you may need to delete your &#39;ScannedFiles.db&#39; file in VDF folder.&lt;br&gt;
Last build based on&lt;/p&gt;</content>
    <author>
      <name>github-actions[bot]</name>
    </author>
    <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/in/15368?s=60&amp;v=4"/>
  </entry>
  <entry>
    <id>tag:github.com,2008:Repository/164539637/2.0.8</id>
    <updated>2020-09-03T18:51:39Z</updated>
    <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.8"/>
    <title>Video Duplicate Finder 2.0.8</title>
    <content>No content.</content>
    <author>
      <name>0x90d</name>
    </author>
    <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
  </entry>
  <entry>
    <id>tag:github.com,2008:Repository/164539637/2.0.7</id>
    <updated>2019-04-30T11:06:32Z</updated>
    <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.7"/>
    <title>Video Duplicate Finder 2.0.7</title>
    <content>No content.</content>
    <author>
      <name>0x90d</name>
    </author>
    <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
  </entry>
  <entry>
    <id>tag:github.com,2008:Repository/164539637/2.0.6</id>
    <updated>2019-02-21T15:10:34Z</updated>
    <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.6"/>
    <title>Video Duplicate Finder 2.0.6</title>
    <content>No content.</content>
    <author>
      <name>0x90d</name>
    </author>
    <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
  </entry>
  <entry>
    <id>tag:github.com,2008:Repository/164539637/2.0.5</id>
    <updated>2019-02-19T18:35:47Z</updated>
    <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.5"/>
    <title>2.0.5</title>
    <content type="html">&lt;p&gt;Assembly and publish profile cleanup - drop 32bit build&lt;/p&gt;</content>
    <author>
      <name>0x90d</name>
    </author>
    <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
  </entry>
  <entry>
    <id>tag:github.com,2008:Repository/164539637/2.0.4</id>
    <updated>2019-01-25T11:58:31Z</updated>
    <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.4"/>
    <title>Video Duplicate Finder 2.0.4</title>
    <content type="html">&lt;p&gt;Hotfix for wrong / missing video height information&lt;/p&gt;</content>
    <author>
      <name>0x90d</name>
    </author>
    <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
  </entry>
  <entry>
    <id>tag:github.com,2008:Repository/164539637/2.0.3</id>
    <updated>2019-01-24T15:46:18Z</updated>
    <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.3"/>
    <title>2.0.3</title>
    <content type="html">&lt;p&gt;Update translations&lt;/p&gt;</content>
    <author>
      <name>0x90d</name>
    </author>
    <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
  </entry>
  <entry>
    <id>tag:github.com,2008:Repository/164539637/2.0.2</id>
    <updated>2019-01-14T20:21:19Z</updated>
    <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.2"/>
    <title>2.0.2</title>
    <content type="html">&lt;p&gt;Console - return on database cleanup&lt;/p&gt;</content>
    <author>
      <name>0x90d</name>
    </author>
    <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
  </entry>
  <entry>
    <id>tag:github.com,2008:Repository/164539637/2.0.1</id>
    <updated>2019-01-12T19:31:24Z</updated>
    <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.1"/>
    <title>Video Duplicate Finder 2.0.1</title>
    <content>No content.</content>
    <author>
      <name>0x90d</name>
    </author>
    <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
  </entry>
  <entry>
    <id>tag:github.com,2008:Repository/164539637/2.0.x</id>
    <updated>2019-01-12T14:41:35Z</updated>
    <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.x"/>
    <title>2.0.x</title>
    <content type="html">&lt;p&gt;Update README.md&lt;/p&gt;</content>
    <author>
      <name>0x90d</name>
    </author>
    <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
  </entry>
</feed>
```

Let's check it in some days

**Comment by rezad1393** [comment]
> Let's do it structured.
> 
> This is a snapshot from now:
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <feed xmlns="http://www.w3.org/2005/Atom" xmlns:media="http://search.yahoo.com/mrss/" xml:lang="en-US">
>   <id>tag:github.com,2008:https://github.com/0x90d/videoduplicatefinder/releases</id>
>   <link type="text/html" rel="alternate" href="https://github.com/0x90d/videoduplicatefinder/releases"/>
>   <link type="application/atom+xml" rel="self" href="https://github.com/0x90d/videoduplicatefinder/releases.atom"/>
>   <title>Release notes from videoduplicatefinder</title>
>   <updated>2021-08-17T11:20:51Z</updated>
>   <entry>
>     <id>tag:github.com,2008:Repository/164539637/3.0.x</id>
>     <updated>2024-05-07T11:38:48Z</updated>
>     <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/3.0.x"/>
>     <title>3.0.x</title>
>     <content type="html">&lt;p&gt;Note: These are daily builds. Attachments are automatically created and replaced on every new commit. Regardless of what the message above says. If you experience upgrade issues you may need to delete your &#39;ScannedFiles.db&#39; file in VDF folder.&lt;br&gt;
> Last build based on&lt;/p&gt;</content>
>     <author>
>       <name>github-actions[bot]</name>
>     </author>
>     <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/in/15368?s=60&amp;v=4"/>
>   </entry>
>   <entry>
>     <id>tag:github.com,2008:Repository/164539637/2.0.8</id>
>     <updated>2020-09-03T18:51:39Z</updated>
>     <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.8"/>
>     <title>Video Duplicate Finder 2.0.8</title>
>     <content>No content.</content>
>     <author>
>       <name>0x90d</name>
>     </author>
>     <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
>   </entry>
>   <entry>
>     <id>tag:github.com,2008:Repository/164539637/2.0.7</id>
>     <updated>2019-04-30T11:06:32Z</updated>
>     <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.7"/>
>     <title>Video Duplicate Finder 2.0.7</title>
>     <content>No content.</content>
>     <author>
>       <name>0x90d</name>
>     </author>
>     <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
>   </entry>
>   <entry>
>     <id>tag:github.com,2008:Repository/164539637/2.0.6</id>
>     <updated>2019-02-21T15:10:34Z</updated>
>     <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.6"/>
>     <title>Video Duplicate Finder 2.0.6</title>
>     <content>No content.</content>
>     <author>
>       <name>0x90d</name>
>     </author>
>     <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
>   </entry>
>   <entry>
>     <id>tag:github.com,2008:Repository/164539637/2.0.5</id>
>     <updated>2019-02-19T18:35:47Z</updated>
>     <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.5"/>
>     <title>2.0.5</title>
>     <content type="html">&lt;p&gt;Assembly and publish profile cleanup - drop 32bit build&lt;/p&gt;</content>
>     <author>
>       <name>0x90d</name>
>     </author>
>     <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
>   </entry>
>   <entry>
>     <id>tag:github.com,2008:Repository/164539637/2.0.4</id>
>     <updated>2019-01-25T11:58:31Z</updated>
>     <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.4"/>
>     <title>Video Duplicate Finder 2.0.4</title>
>     <content type="html">&lt;p&gt;Hotfix for wrong / missing video height information&lt;/p&gt;</content>
>     <author>
>       <name>0x90d</name>
>     </author>
>     <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
>   </entry>
>   <entry>
>     <id>tag:github.com,2008:Repository/164539637/2.0.3</id>
>     <updated>2019-01-24T15:46:18Z</updated>
>     <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.3"/>
>     <title>2.0.3</title>
>     <content type="html">&lt;p&gt;Update translations&lt;/p&gt;</content>
>     <author>
>       <name>0x90d</name>
>     </author>
>     <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
>   </entry>
>   <entry>
>     <id>tag:github.com,2008:Repository/164539637/2.0.2</id>
>     <updated>2019-01-14T20:21:19Z</updated>
>     <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.2"/>
>     <title>2.0.2</title>
>     <content type="html">&lt;p&gt;Console - return on database cleanup&lt;/p&gt;</content>
>     <author>
>       <name>0x90d</name>
>     </author>
>     <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
>   </entry>
>   <entry>
>     <id>tag:github.com,2008:Repository/164539637/2.0.1</id>
>     <updated>2019-01-12T19:31:24Z</updated>
>     <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.1"/>
>     <title>Video Duplicate Finder 2.0.1</title>
>     <content>No content.</content>
>     <author>
>       <name>0x90d</name>
>     </author>
>     <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
>   </entry>
>   <entry>
>     <id>tag:github.com,2008:Repository/164539637/2.0.x</id>
>     <updated>2019-01-12T14:41:35Z</updated>
>     <link rel="alternate" type="text/html" href="https://github.com/0x90d/videoduplicatefinder/releases/tag/2.0.x"/>
>     <title>2.0.x</title>
>     <content type="html">&lt;p&gt;Update README.md&lt;/p&gt;</content>
>     <author>
>       <name>0x90d</name>
>     </author>
>     <media:thumbnail height="30" width="30" url="https://avatars.githubusercontent.com/u/46010672?s=60&amp;v=4"/>
>   </entry>
> </feed>
> ```
> 
> Let's check it in some days

again, it doesnt update to newest version.

![Screenshot 2024-05-11 at 07-13-08 videoduplicatefinder · FreshRSS](https://github.com/FreshRSS/FreshRSS/assets/7523164/dc068736-0ea3-4b56-b250-bd1da0bf23c5)


this is my feed in frehsrss.
when I manually press update feed on feed option it doesnt find new (updated date     <updated>2024-05-07T11:38:48Z</updated> ) for that top post.

**Reply by math-GH** [reply]
I understand your frustration. Please keep relaxed. Concrete data helps to understand the problem. There are algorithms working that tries do analyze if the feed changed or not.

**Reply by rezad1393** [reply]
oh,was my tone angry or frustrated?

I am happy that you are helping with this and the lack of info from my part is what caused the first misunderstanding.

it is only one feed and I am not in a hurry and this is a free software and I mostly report issues to make software better and not really expect servitude from volunteers helping.

take your time.

I only reported this because I have a concrete bug that I can reproduce (old post that doesn't get updated) and want to make freshrss better.

usually my issue reporting is bad and not really concrete.

**Comment by Alkarex** [comment]
At the moment, we do not use the `updated` field for detecting a change, but rather check whether there has been a change in the content:

https://github.com/FreshRSS/FreshRSS/blob/fa731db2861fef5cd0911a1566bed4328a12ab78/app/Models/Entry.php#L437

We could store that information into the entry attributes via SimplePie's `get_updated_date()` function.
Edit: in this case, since there is no other date, just checking whether the date has changed should be enough.

We should however test whether that does not trigger too many false positives. We indeed typically have the opposite problems, namely of marking as unread too often...

**Reply by Frenzie** [reply]
I think in a GitHub context it could mean a new artifact was added. But if that's not reflected in the feed content it'd definitely feel like a false positive even if it strictly speaking wasn't.

**Comment by rezad1393** [comment]
> I think in a GitHub context it could mean a new artifact was added. But if that's not reflected in the feed content it'd definitely feel like a false positive even if it strictly speaking wasn't.

I am not familiar with artifact but that repo create a new binary every time a commit is added but instead of a new rss post it just replaces the last release

**Comment by rezad1393** [comment]
> At the moment, we do not use the `updated` field for detecting a change, but rather check whether there has been a change in the content:
> 
> https://github.com/FreshRSS/FreshRSS/blob/fa731db2861fef5cd0911a1566bed4328a12ab78/app/Models/Entry.php#L437
> 
> We could store that information into the entry attributes via SimplePie's `get_updated_date()` function. Edit: in this case, since there is no other date, just checking whether the date has changed should be enough.
> 
> We should however test whether that does not trigger too many false positives. We indeed typically have the opposite problems, namely of marking as unread too often...

I would have thought that it was the opposite, you know, first checking the date.

do feeds content change without date change?

doesnt rss software run on the host that creates rss feeds usually  updates the date automatically when the author changes the post content??

**Comment by math-GH** [comment]
Here is a solution, that solves your issue too:

`commits.atom` has all commits into the main branch.

For FreshRSS: https://github.com/FreshRSS/FreshRSS/commits.atom
For your videoduplicatefinder: https://github.com/0x90d/videoduplicatefinder/commits.atom

It is much more useful than watching this strange used release RSS feed.

**Reply by rezad1393** [reply]
I have already done that.

I just wanted to help freshrss get better.
as I said before.
